### PR TITLE
Added a Function To Get A T-Value From Arclength In Firmware Polynomial2d

### PR DESCRIPTION
**Pull request by garethellis0**
### Testing Done
Unit tests.

### Resolved Issues
N/A

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by EvanMorcom — DISMISSED** [review]
That was fast!

LGTM 🔥

File: src/firmware/main/math/polynomial_2d_test.cpp [hunk 1]
```diff
@@ -352,3 +351,112 @@ TEST_F(Polynomial2dTest, get_position_at_arc_length_on_complex_line_multiple_div
...
+{
+    // This function should probably never be used like this, but this checks that
+    // we're at least executing defined behavior
+
+    Polynomial2dOrder1_t poly = {
+        .x = {.coefficients = {1, 0.1}},
+        .y = {.coefficients = {1, 0.2}},
+    };
+
+    static float t_values[1]                              = {0.5};
```
Comment (MathewMacDougall) [inline src/firmware/main/math/polynomial_2d_test.cpp]: Who do these need to be `static`?

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]

File: src/firmware/main/math/polynomial_2d_test.cpp (hunk 1 above)
Comment (garethellis0) [reply src/firmware/main/math/polynomial_2d_test.cpp]: Whoops, they don't. Removed.

**Review by MathewMacDougall — APPROVED** [review]

**Review by EvanMorcom — APPROVED** [review]
Beauty

**Pull request merged**